demos/sample_azure_iot_ca_recovery/az-function/recovery.csx

```diff
@@ -0,0 +1,232 @@
+"YaRSMLl6IE9vDzhTyzMMEyuP1pq9KsgtsRx9S1HKR9FIJ3Jdh+vVReZIZZ2vUpC6\r\n" +
+"W6IYZVcSn2i51BVrlMRpIpj0M+Dt+VGOQVDJNE92kKz8OMHY4Xu54+OU4UZpyw4K\r\n" +
+"UGsTuqwPN1q3ErWQgR5WrlcihtnJ0tHXUeOrO8ZV/R4O03QK0dqq6mm4lyiPSMQH\r\n" +
+"+FJDOvTKVTUssKZqwJz58oHhEmrARdlns87/I6KJClTUFLkqqNfs+avNJVgyeY+Q\r\n" +
+"W5g5xAgGwax/Dj0ApQIDAQABo1QwUjAOBgNVHQ8BAf8EBAMCAYYwDwYDVR0TAQH/\r\n" +
+"BAUwAwEB/zAdBgNVHQ4EFgQUCctZf4aycI8awznjwNnpv7tNsiMwEAYJKwYBBAGC\r\n" +
+"NxUBBAMCAQAwDQYJKoZIhvcNAQEMBQADggIBAKyvPl3CEZaJjqPnktaXFbgToqZC\r\n" +
+"LgLNFgVZJ8og6Lq46BrsTaiXVq5lQ7GPAJtSzVXNUzltYkyLDVt8LkS/gxCP81OC\r\n" +
+"gMNPOsduET/m4xaRhPtthH80dK2Jp86519efhGSSvpWhrQlTM93uCupKUY5vVau6\r\n" +
+"tZRGrox/2KJQJWVggEbbMwSubLWYdFQl3JPk+ONVFT24bcMKpBLBaYVu32TxU5nh\r\n" +
+"SnUgnZUP5NbcA/FZGOhHibJXWpS2qdgXKxdJ5XbLwVaZOjex/2kskZGT4d9Mozd2\r\n" +
+"TaGf+G0eHdP67Pv0RR0Tbc/3WeUiJ3IrhvNXuzDtJE3cfVa7o7P4NHmJweDyAmH3\r\n" +
+"pvwPuxwXC65B2Xy9J6P9LjrRk5Sxcx0ki69bIImtt2dmefU6xqaWM/5TkshGsRGR\r\n" +
+"xpl/j8nWZjEgQRCHLQzWwa80mMpkg/sTV9HB8Dx6jKXB/ZUhoHHBk2dxEuqPiApp\r\n" +
+"GWSZI1b7rCoucL5mxAyE7+WL85MB+GqQk2dLsmijtWKP6T+MejteD+eMuMZ87zf9\r\n" +
+"dOLITzNy4ZQ5bb0Sr74MTnB8G2+NszKTc0QWbej09+CVgI+WXTik9KveCjCHk9hN\r\n" +
+"AHFiRSdLOkKEW39lt2c0Ui2cFmuqqNh7o0JMcccMyj6D5KbvtwEwXlGjefVwaaZB\r\n" +
+"RA+GsCyRxj3qrg+E\r\n" +
+"-----END CERTIFICATE-----\r\n";
+
+public static async Task<IActionResult> Run(HttpRequest req, ILogger log) 
+
+{ 
+    log.LogInformation("C# HTTP trigger function processed a request.");
+    // Get request body 
+    string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+
+    dynamic? data = JsonConvert.DeserializeObject(requestBody);
+
+    log.LogInformation("Request.Body:...");
+    log.LogInformation(requestBody);
+
+    // Get registration ID of the device 
+    string? regId = data?.deviceRuntimeContext?.registrationId;
+    string message = "Uncaught error";
+    bool fail = false;
+    ResponseObj obj = new ResponseObj();
+
+    if (regId == null) 
+    {
+        message = "Registration ID not provided for the device.";
+        log.LogError("Registration ID : NULL");
+        fail = true;
+    }
+    else
+    {
+        string?[] hubs = data?.linkedHubs?.ToObject<string[]>();
+
+        // Must have hubs selected on the enrollment 
+        if (hubs == null || hubs.Length == 0) 
+        {
+            message = "No hub group defined for the enrollment.";
+            log.LogError("linkedHubs : NULL");
+            fail = true;
+        }
+        else 
+        {
+            message = "Succeeded.";
+            obj.iotHubHostName = hubs[0];
+        }
+    }
+    // create cert bundle payload
+    ResponsePayload p = new ResponsePayload();
+
+    TrustBundlePayload certTrustBundle = new TrustBundlePayload();
+    certTrustBundle.certs  = certTrust1;
+    certTrustBundle.version = certTrust1Version;
+    // Calc epoch time
+    TimeSpan t = DateTime.UtcNow - new DateTime(1970, 1, 1);
```

Comment:
Nit : [DateTimeOffset.ToUnixTimeSeconds](https://learn.microsoft.com/en-us/dotnet/api/system.datetimeoffset.tounixtimeseconds?view=net-7.0)

Reply:
looks like the end usage is roughly similar so I'll leave as is for now.